ui/static/ui/js/learning_resources.jsx

```diff
@@ -6,73 +6,160 @@ define('learning_resources', [
+            placeholder={"Select a value for " + this.props.selectedVocabulary.name}
             options={options}
             onChange={this.handleChange}
-            values={this.props.selectedTerms}
-            multiple={this.props.vocabulary.multi_terms}
+            values={currentVocabulary.selectedTerms}
+            multiple={this.props.selectedVocabulary.multi_terms}
             />
-        </div>
-      </div>;
+        </div>;
     },
+
     handleChange: function(e) {
-      var selectValues = _.pluck(
+      var selectedValues = _.pluck(
         _.filter(e.target.options, function(option) {
           return option.selected && option.value !== null;
         }), 'value');
-      this.props.updateTerms(this.props.vocabulary.slug, selectValues);
+
+      this.props.updateTerms(this.props.selectedVocabulary.slug, selectedValues);
     }
   });
 
   var LearningResourcePanel = React.createClass({
     mixins: [React.addons.LinkedStateMixin],
-    render: function () {
+    
+    updateVocabulary: function(vocabs, selectedValue){
       var thiz = this;
-      var vocabulariesAndTerms = this.state.vocabulariesAndTerms;
-      var options = _.map(vocabulariesAndTerms, function (pair) {
-        var updateTerms = function(vocabSlug, newTermsSlug) {
-          var newVocabulariesAndTerms = _.map(vocabulariesAndTerms,
-            function(tuple) {
-              if (vocabSlug === tuple.vocabulary.slug) {
-                return {
-                  vocabulary: tuple.vocabulary,
-                  terms: tuple.terms,
-                  selectedTerms: newTermsSlug
-                };
-              } else {
-                return tuple;
-              }
-            });
 
-          thiz.setState({
-            vocabulariesAndTerms: newVocabulariesAndTerms
-          });
-        };
+      var selectedVocabulary = _.find(vocabs, _.matchesProperty('slug', selectedValue));
 
-        return <VocabularyOption
-          vocabulary={pair.vocabulary}
-          terms={pair.terms}
-          selectedTerms={pair.selectedTerms}
-          key={pair.vocabulary.slug}
-          updateTerms={updateTerms}
-          />;
+      thiz.setState({
+        selectedVocabulary: selectedVocabulary
       });
+    },
+
+    updateTerms: function(vocabSlug, selectedTerms){
+      // QUESTION: why do we have to pass a reference to the LP Component into the sub-component?
+      var thiz = this;
+      var vocabulariesAndTerms = thiz.state.vocabulariesAndTerms;
+
+      var newVocabulariesAndTerms = _.map(vocabulariesAndTerms,
+        function(tuple) {
+          if (vocabSlug === tuple.vocabulary.slug) {
+            var newTuple = {
+              vocabulary: tuple.vocabulary,
+              terms: tuple.terms,
+              selectedTerms: selectedTerms
+            };
+            return newTuple;
+          } else {
+            return tuple;
+          }
+        });
+
+      thiz.setState({
```

Comment:
I don't think you should need to use `thiz` for these unless you need to access the `this` from within a closure.